### PR TITLE
New Source - Calendly

### LINKED ISSUE #3284: Calendly
### App
`calendly_v2`
https://developer.calendly.com/

### Triggers
- Event created
- Event canceled

### PR DESCRIPTION
**Pull request by vellames-turing**
Closes #3284

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3449"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/calendly_v2/sources/invitee-action/invitee-action.mjs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  hooks: {
+    async activate() {
+      const previousWebhook = this._getWebhook();
+      if (previousWebhook) {
+        await this._deactivateWebhook(previousWebhook);
+      }
+
+      await this._registerWebhook();
+    },
+  },
```
Comment (feyzullah) [inline components/calendly_v2/sources/invitee-action/invitee-action.mjs:136]: Don't we need a `deactivate` function here? How is deactivated webhook deleted from calendly? Please excuse me if I miss something here.

File: components/calendly_v2/sources/invitee-action/invitee-action.mjs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    } = this._getCalendlySignature(event);
+    if (!t || !signature) throw new Error("Invalid Signature");
+
+    this._checkHmac(t, signature, event.bodyRaw);
+    this._checkReplayAttack(t);
+
+    this.$emit(event.body.payload, {
+      id: event.body.payload.uri,
+      summary: event.body.payload.name,
+      ts: Date.now(),
```
Comment (feyzullah) [inline components/calendly_v2/sources/invitee-action/invitee-action.mjs:152]: If event body has a date, can we use it instead of `Date.now()`?

**Review by feyzullah — COMMENTED** [review]
Hi @vellames-turing, I added just a few comments. Can you please check them?

File: components/calendly_v2/sources/invitee-action/invitee-action.mjs (hunk 1 above)
Comment (vellames-turing) [reply components/calendly_v2/sources/invitee-action/invitee-action.mjs:136]: Oops!! I missed it

File: components/calendly_v2/sources/invitee-action/invitee-action.mjs (hunk 2 above)
Comment (vellames-turing) [reply components/calendly_v2/sources/invitee-action/invitee-action.mjs:152]: Hmmm, I usually use `ts` as the date we emitted the event. I already saw people using `Date.now()` and the event date, maybe this is a good opportunity to clarify.
@dannyroosevelt Just confirming, what is the best approach here? Use the current time or the event time?

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I requested just a some little changes.

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification Completed, and this is ready for release.

**Review by lcaresia — APPROVED** [review]

**Pull request merged**